wordpress/wp-content/plugins/memberful-wp/src/admin.php

```diff
@@ -414,48 +414,72 @@ function memberful_wp_advanced_settings() {
 function memberful_wp_bulk_protect() {
   if ( ! empty( $_POST ) ) {
     $categories_to_protect = empty( $_POST['memberful_protect_categories'] ) ? array() : (array) $_POST['memberful_protect_categories'];
+    $categories_to_protect = array_map( 'intval', $categories_to_protect );
+    $categories_to_protect = array_intersect( $categories_to_protect, get_categories( array( 'fields' => 'ids' ) ) );
+
     $acl_for_products = empty( $_POST['memberful_product_acl'] ) ? array() : (array) $_POST['memberful_product_acl'];
+    $acl_for_products = array_map( 'intval', $acl_for_products );
+    $acl_for_products = array_intersect( $acl_for_products, array_keys( memberful_downloads() ) );
+
     $acl_for_subscriptions = empty( $_POST['memberful_subscription_acl'] ) ? array() : (array) $_POST['memberful_subscription_acl'];
-    $marketing_content = empty( $_POST['memberful_marketing_content'] ) ? '' : $_POST['memberful_marketing_content'];
-    $things_to_protect = empty( $_POST['target_for_restriction']) ? '' : $_POST['target_for_restriction'];
-    $viewable_by_any_registered_user = empty( $_POST['memberful_viewable_by_any_registered_users'] ) ? '' : $_POST['memberful_viewable_by_any_registered_users'];
-    $viewable_by_anybody_subscribed_to_a_plan = empty( $_POST['memberful_viewable_by_anybody_subscribed_to_a_plan'] ) ? '' : $_POST['memberful_viewable_by_anybody_subscribed_to_a_plan'];
+    $acl_for_subscriptions = array_map( 'intval', $acl_for_subscriptions );
+    $acl_for_subscriptions = array_intersect( $acl_for_subscriptions, array_keys( memberful_subscription_plans() ) );
+
+    $marketing_content = trim( wp_kses_post( $_POST['memberful_marketing_content'] ) );
+    $viewable_by_any_registered_user = isset( $_POST['memberful_viewable_by_any_registered_users'] );
+    $viewable_by_anybody_subscribed_to_a_plan = isset( $_POST['memberful_viewable_by_anybody_subscribed_to_a_plan'] );
 
     $subscription_acl_manager = new Memberful_Post_ACL( Memberful_ACL::SUBSCRIPTION );
     $product_acl_manager = new Memberful_Post_ACL( Memberful_ACL::DOWNLOAD );
 
     $query_params = array('nopaging' => true, 'fields' => 'ids');
 
-    switch ( $things_to_protect ) {
-    case 'all_pages_and_posts':
-      $query_params['post_type'] = array('post', 'page');
-      break;
-    case 'all_pages':
-      $query_params['post_type'] = 'page';
-      break;
-    case 'all_posts':
-      $query_params['post_type'] = 'post';
-      break;
-    case 'all_posts_from_category':
-      $query_params['category__in'] = $categories_to_protect;
-      break;
-    default:
-      $query_params['post_type'] = $things_to_protect;
+    if ( isset( $_POST['target_for_restriction'] ) ) {
```

Comment:
Non-blocking. WDYT of moving this check to the beginning of the method and calling `wp_die` if it's not set? I'm finding these nested conditions a bit hard to follow.

Reply:
I slightly prefer to keep it as it is, so all `$_POST['target_for_restriction`]` checks are together.

But I'm open to refactoring it later.